### DISCUSSION TITLE
Missing basic multistore feature on PS 8

### DISCUSSION THREAD

**Discussion by cerodesign** [PrestaShop/PrestaShop#34225] [Ideas]
### The problem

![PS16](https://github.com/PrestaShop/PrestaShop/assets/42432086/c9dba06a-71e0-4531-aa93-f5b930c51546)
On old Prestashop 1.6 there were the frature on multistore product eddting, which allow you to edit just one thing on product. forexample you want to edit just price. so you chceked the chceckbox beside the Pre-tax retail price, edit the price  and save it. and you affecet just he price.
 But new Presta 8 completely mis this frature, everitime you want to eddit just one thing on produkt, you revrite all data on all mulstistores.   

### The solution

give the old feature with  chceckboxs back. 

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by cerodesign** [comment]
AND ... the new PS 8.1 has this ability again (since my test was on PS 8.0.3) however it miss other ability. In PS 8.1, you are not possible to use to add product to all stores, just to one of them. And than you need to add the produkt to another, and another, which completely degrate the multisore purpose (to add product just onetime and have it in all shops), since you have to add the product to every singe eshop!

**Comment by cerodesign** [comment]
I reportet this as a BUG and it shopld be repaired on PS 8.1.2, hopefully.

**Comment by MatShir** [comment]
Hi @cerodesign, 
I wanted to share with you the latest update of the product page on PrestaShop 8.1.

For security reason, you can't edit the product in 'all store'. It was causing too much error with big impacts. Now, to avoid editing each page, you have a checkbox that allows you to add the changes to all the stores.

![Capture d’écran 2023-10-10 à 11 55 42](https://github.com/PrestaShop/PrestaShop/assets/43613217/f295ea48-ecba-4038-ba8d-afbfa33f6595)

And for the images, you can select directly the images you want to be available for each shops.

![Capture d’écran 2023-10-10 à 11 56 18](https://github.com/PrestaShop/PrestaShop/assets/43613217/b06480f8-ed94-46d0-8ab8-689f3b2d7d60)

And so you can create one product for different stores, you have to assign the stores once the product is created. Press the 'select stores' button 

![Capture d’écran 2023-10-10 à 11 56 45](https://github.com/PrestaShop/PrestaShop/assets/43613217/6e06d60b-d083-46ae-9cfe-96bc4b43fde8)

Then select the store you want to assign it to

![Capture d’écran 2023-10-10 à 11 56 52](https://github.com/PrestaShop/PrestaShop/assets/43613217/285fcc87-da1b-4e66-b14c-0e7e8071ceba)
 
 Maybe we can improve the userflow, so you can find it more easily. cc @TristanLDD @ElodieOS

**Comment by cerodesign** [comment]
Thanks. I work in PS from prestashop 1.4, but I was not able to find out this abbility 

 

Ok thank I will tried it. 

 

Thank you. 

 

S pozdravom

 

--

**Comment by darkWolf-PR** [comment]
@MatShir hi, I just went through the update from PS 1.6 and find out this pretty odd workflow. I´m pretty worried about BFUs that work with our eshops, if they actually manage to understand, how this works after using 1.6 for years. 

One thing that I do not really understand - WHY is the Product listing page showing 0 ( ZERO ) quantity, if we have set multistore to share quantity in group? It has to be same for the product on all stores.

We use multistore only because of this feature, because of sharing quantities. All stores has same products, same categories and if the product is sold on one store, it has to be set on zero quantity in all the others too. Otherwise, we would be selling stuff we don´t have...

**Reply by MatShir** [reply]
Hi, 
So the share quantity are set on the group shop level. So if the product is associated to the shops from the same shop group and set share stock, it should work. I did the test, it worked fine. 
Could you provide more info to reproduce the bugs and on which version you are? 

But the products are not "shared". When the multistore is set, the products are duplicated from store to store. So when you create a product on the first shop make sure to assign it to the other shop. 

> hi, I just went through the update from PS 1.6 and find out this pretty odd workflow. I´m pretty worried about BFUs that work with our eshops, if they actually manage to understand, how this works after using 1.6 for years.

Not sure to understand, could you detail which workflow you are reffering to ?

**Reply by darkWolf-PR** [reply]
We have multistore with 4 shops, all in same group, set to share stocks (not users or orders), just stocks. It´s latest PS 8.1.3, updated from 1.6.1.24. (jump from 1.6 to 1.7.8.7 to 8.1.3) and if I open catalog, all products shows quantity 0, I have to click on the arrow that opens listing of the product on all shops and here the real quantity is shown, same for all shops.

I did tried to make new product, not migrated one and it´s the same for that. Maybe I did not say it right, but I meant when u have set context to "all shops" in catalog. The same way we had it several years in PS 1.6. It shows number, if u set context to one of the shops, but as set it to "all shops", its showing zeros for the stock. 

I know they are effectively copied, but the 1.6 sems a lot more logical for using, at least for people, who have no idea how it actually works.

Situation - we are not actually creating products. We are creating products(name, product code, price) in accounting software, these products are then copied by special bridge to our PS. After that, eshop stuff starts adding images, texts etc., That means they do miss the part of creating a product, which gives u the idea, how it works and pushes u into single store context. I´m actually pretty worried of possible problems, before, the just had set "all shops" all the time and everything worked, now, PS will be switching em context and they will be forgetting on it, which will be leading to errors, missing content etc.

**Reply by MatShir** [reply]
There was clearly a bug with the stock. Initially, the default shop was displaying the quantity for the name, price, and category. I see it now. Should we create a bug report? I will prioritize it for the next patch, as it doesn't seem too difficult to fix.

I don't fully agree, but maybe a quick call would be easier to discuss your use case. We received many complaints about merchants editing content on 'all shops' without realizing that the context was 'all shops', resulting in terrible consequences when they wanted to edit in group or single shops. That's why we reversed the logic. Now the edition will be made in a single shop context, but we can apply it everywhere if you want, and there is a checkbox for it.
Would you like to discuss on a call to better understand your concerns and use cases? (join the slack, you can find me under MatShir)

**Reply by cerodesign** [reply]
Yes you were right, it is a bug. 
and the is 2 others bugs (in PS 8.1.3, 8.1.2 ....)
1. when you establish mutistore (for example 2 stores) you have 2 produkts of all produkts in store. What is logical. And then you want to edit a produkt and after edit the system make 3 produkt of this edited produt. 2x for primary store. what for? this is reali misleading. see attachment
2. when you try to make a product with combination,  the system do not makr a one combination as a default one, so it makes the product not able to sell, when someone open it in frontoffice. This is just a problem of multistore, on normal store (not a multistore) system autoamticky choose first combination as default one. see attachment

To tell the true: "golden PS 1.6. "

Carries:
And last realy realy nonsensical issue in presrashop (it is from PS 1.4 ... till now) , but this is not connected to multistore, is that  why Prestashop always, when you eddit something in carrie, change the ID of carrier. This realy make more problems than you think, and it does not make sence.    
  
![before-edit](https://github.com/PrestaShop/PrestaShop/assets/42432086/10e2429b-5c7d-4454-86f9-0748fbf51b48)
![after-edit](https://github.com/PrestaShop/PrestaShop/assets/42432086/9d0a1bd4-0700-4a7f-a5ce-b057e87f0fa2)
![default-comb](https://github.com/PrestaShop/PrestaShop/assets/42432086/02114883-e3af-44f6-bbf7-600f9c8f46f0)

**Reply by darkWolf-PR** [reply]
@cerodesign We have this 3x after edit, for the 1st shop, product shows twice, one activated and one deactivated with zero stock.

@MatShir that would be nice to get fixed.

Also, when editing, we found out there is missing "all shops" checkbox for "Default category", so in case we want to change default category(we do have same categories, same category root for all) for all shops, it´s impossible(someone set´s the wrong one first or whatever). When u make new product, default category saves to all shops, but if u want to change it after, and again save to all-it doesn´t work. For us, this means we will have to update default category for every product 4x, because as we are using bridge from accounting software to set up products, they are set up with default root category.
